### PR TITLE
chore(ButtonDeprecated): Dead Code removal - unused themes [GM-166]

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

Just cleaning up some unused CSS from buttons here and getting us in better shape for migration and save a bit on chunk size for buttons while we migrate.

Monolith usage statistics (`theme="[^"](.*)"`):
```tsx
const currentUsage = {
  "brand-blue": 42,
  grey: 2,
  "brand-purple": 37,
  "brand-dark-blue": 8,
  white: 31,
  "brand-yellow": 26,
  lantern: 10,
  secondary: 5,
  red: 5,
  platform: 4,
  blue: 1,
  yellow: 1,
  royalblue: 3,
  "brand-white": 1,
  navy: 1,
  hyper: 1,
  "brand-red": 1,
};
```

Removals:
- "brand-pink"
- "brand-mind"
- "brand-beige"
- "brand-orange"
-  "purple"
- "notice"
- "error"
- "announcement"
- "success"
- "info"

Mapped:
-  "royalblue" => "brand-purple"

- Removes unused button styles to lower generated CSS (Each theme is around ~1500 characters)
- Adds strict theme types to ButtonDeprecated themes to make consolidation going forward easier.
- Updates storybook with new variables

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: GM-166
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/styleguide/stories/Atoms/ButtonDeprecated/helpers.tsx [hunk 1]
```diff
@@ -20,17 +16,7 @@ export const deprecatedThemeKeys = [
   'darkmint',
   'grey',
   'greyblue',
-  'white',
   'royalblue',
-  'purple',
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Atoms/ButtonDeprecated/helpers.tsx:25]: Just confirming on these removals: I also looked for `ButtonDeprecatedProps` in the monolith and found a few prop names like `buttonProps`. None of them had anything not used by other places that I could see: e.g. `ExecrciseNextButton` uses `'brand-yellow'`.

**Review by JoshuaKGoldberg — APPROVED** [review]
🔪

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.17-alpha.7f1922.0<br/>@codecademy/gamut-labs@11.3.1-alpha.7f1922.0<br/>@codecademy/gamut@26.4.1-alpha.7f1922.0<br/>@codecademy/styleguide@30.7.1-alpha.7f1922.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://604fdd0566c4a3102d10ca9b--gamut-preview.netlify.app>https://604fdd0566c4a3102d10ca9b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/604fdd0566c4a3102d10ca9b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.29-alpha.4d6b82.0<br/>@codecademy/gamut-labs@13.0.5-alpha.4d6b82.0<br/>@codecademy/gamut@29.1.1-alpha.4d6b82.0<br/>@codecademy/styleguide@33.0.3-alpha.4d6b82.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60591b503321c700bf9fa399--gamut-preview.netlify.app>https://60591b503321c700bf9fa399--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60591b503321c700bf9fa399>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.31-alpha.5d1845.0<br/>@codecademy/gamut-labs@13.1.1-alpha.5d1845.0<br/>@codecademy/gamut@29.3.1-alpha.5d1845.0<br/>@codecademy/styleguide@33.2.1-alpha.5d1845.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6059ea39c3acce512e13ee30--gamut-preview.netlify.app>https://6059ea39c3acce512e13ee30--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6059ea39c3acce512e13ee30>Deploy Logs</a></p>

**Pull request merged**